paper-api/src/main/java/io/papermc/paper/event/entity/CopperGolemValidateTargetEvent.java

```diff
@@ -0,0 +1,29 @@
+package io.papermc.paper.event.entity;
+
+import org.bukkit.block.Block;
+import org.bukkit.entity.CopperGolem;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Specialization of {@link ItemTransportingEntityValidateTargetEvent} only
+ * called for {@link CopperGolem}s. This event may be more convenient for
+ * certain use cases, but if capturing all interactions is desired
+ * (i.e. for a protection plugin), the parent event should be used instead.
+ */
+@NullMarked
+public class CopperGolemValidateTargetEvent extends ItemTransportingEntityValidateTargetEvent {
```

Comment:
I do not see a real reason to have this as a separate event. If someone only wants `CopperGolem` they can do an `instanceof` check.

Reply:
after further thought the case where you need to consider the golem's oxidization for target validation is not that common... so it will be rare anyways